feat: Enable Support for Teradata Metadata Scanner

DeepakBhatt13 opened the pull request:
This PR adds teradata scanner for Amundsen.
Contribution doesn't affect any existing modules/codebase in the repository.

mgorsk1 commented on databuilder/databuilder/extractor/teradata_metadata_extractor.py:
```diff
@@ -0,0 +1,44 @@
+            {cluster_source} as cluster, 
+            c.DatabaseName as schema, 
+            c.TableName as name, 
+            c.CommentString as description , 
```
whitespace before comma

DeepakBhatt13 replied:
Removed whitespace.

mgorsk1 commented on databuilder/databuilder/extractor/teradata_metadata_extractor.py:
```diff
@@ -0,0 +1,44 @@
+            d.CommentString as col_description, 
+            d.ColumnId as col_sort_order
+            FROM dbc.Tables c, dbc.Columns d
+            WHERE c.DatabaseName = d.DatabaseName AND c.TableName = d.TableName
```
just curious as I don't have much experience with teradata schema - is there a possibility to join by database **id** and table **id** instead of table database **name** and table **name**? Ids are usually indexed and perform better with joins

DeepakBhatt13 replied:
@mgorsk1 I agree with your point but Id's are not present in dbc.Tables and dbc.Columns in Teradata.

mgorsk1 reviewed (commented):
Thanks for the contribution, I've added couple minor comments.

Would you mind also extending this PR so teradata icon is available in the UI? Exemplary PR to achieve this is here: https://github.com/amundsen-io/amundsen/pull/1100

DeepakBhatt13 commented:
> Thanks for the contribution, I've added couple minor comments.
> 
> Would you mind also extending this PR so teradata icon is available in the UI? Exemplary PR to achieve this is here: #1100

@mgorsk1 I have already incorporated changes for teradata icon in this PR.

mgorsk1 commented on databuilder/requirements.txt:
```diff
@@ -24,6 +24,7 @@ unidecode
 Jinja2>=2.10.0,<4
 pandas>=0.21.0,<1.2.0
 responses>=0.10.6
+teradatasqlalchemy==17.0.0.0
```
can this be an optional (extra) dependency provided through `setup.py`?

mgorsk1 commented:
cicd fails, otherwise lgtm @DeepakBhatt13
